Data/models/Bookmark.swift

```diff
@@ -306,13 +342,14 @@ extension Bookmark {
     
     private func updateInternal(customTitle: String?, url: String?, newSyncOrder: String? = nil,
                                 save: Bool = true, sendToSync: Bool = true,
+                                location: SaveLocation = .keep,
                                 context: WriteContext = .new) {
         
         DataController.perform(context: context) { context in
             guard let bookmarkToUpdate = context.object(with: self.objectID) as? Bookmark else { return }
             
             // See if there has been any change
-            if bookmarkToUpdate.customTitle == customTitle && bookmarkToUpdate.url == url && bookmarkToUpdate.syncOrder == newSyncOrder {
+            if bookmarkToUpdate.customTitle == customTitle && bookmarkToUpdate.url == url && bookmarkToUpdate.syncOrder == newSyncOrder, case .keep = location {
```

Comment:
Yes, this is a really weird way to check equality in enums with associated values.

so I can't do simply do `location == .keep` because one of the enum values has associated value(`case new(location: Bookmark?)`)